mux/mux.go

```diff
@@ -0,0 +1,56 @@
+package mux
+
+import (
+	"errors"
+	"io"
+	"net"
+	"time"
+)
+
+// ErrReset is returned when reading or writing on a reset stream.
+var ErrReset = errors.New("stream reset")
+
+// Stream is a bidirectional io pipe within a connection.
+type MuxedStream interface {
+	io.Reader
+	io.Writer
+
+	// Close closes the stream for writing. Reading will still work (that
+	// is, the remote side can still write).
+	io.Closer
```

Comment:
here it should be reflected in the method name that it closes stream for writing

Reply:
that's a breaking change and we'd like to avoid those.

Reply:
Good point, but yeah, we can't rename _methods_ as that would create breakage, but we can rename _types_. We could consider this for iteration 3 (small breakages).

Reply:
Added it to the manifest for iteration 3 (in the PR description).